### ISSUE TITLE
application of commercial detection : mute the TV during ads.

### ISSUE THREAD

**Issue by ghost** [vasanthkalingeri/CommercialDetection#6]
Greetings,

I am curious if it is possible to use this project as a library for embedded projects such as arduino, etc. 
It would make an interesting project to create an "ad-blocker" spinoff for over the air TV.  @vasanthkalingeri what are you thoughts for processing a live stream using this library?  It would be non-trivial to issue IR "mute" and "un-mute" commands given active commercial detection.

Thanks,
coast

**Comment by vasanthkalingeri** [comment]
Hi coast,

It can be used on low-level devices, it was tested with a Raspberry PI. It would require some work but it could be modified to process live streams with a 5-second delay(Dejavu requirements for a good fingerprint), I have not open-sourced the real-time version yet since there are still some bugs with FFmpeg transcoding.

**Comment by ghost** [comment]
this sounds very interesting, do you plan to free the code modified to process live streams?

**Comment by vasanthkalingeri** [comment]
Yes as soon as I get the transcoder working though :), currently, the code is able to process live streams(on a chunk by chunk basis), but I haven't written parts that can work directly with live stream files(like HLS), so I guess releasing my code now wouldn't be that useful.

**Comment by ghost** [comment]
Thanks for the update!

**Comment by YurieCo** [comment]
@vasanthkalingeri  if u need contribute let me help by the way may I fork it or add ideas on test branch ?

**Comment by ghost** [comment]
Any thoughts with respect to this ML application towards live video streams?

**Comment by YurieCo** [comment]
@mancoast yes, it is recursive if we can auto detect ads then use this data to feed the algorithms which detects  algorithms. kindly recursive (like Reinforcement learning0

**Comment by ZeeshanAhmed92** [comment]
Hi
there is no main.py file to run
i have audios of news, i want to detect ads in it
can you help me ?
